.github/workflows/check_code.yml

```diff
@@ -31,20 +30,19 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: ["3.8", "3.9", "3.10"]
```

Comment:
I think we should keep the python3.6 integration. Even if it's EOL, the the 3.6 version is still available as github actions: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json 

Removing it will break the code of some users of the library that are still on python3.6. Such a change must be done on a separate PR because it's a breaking change.  

However, you can keep the 3.10 integration since it's a new feature

Reply:
You can rebase the master branch to yours. A fix have been added for the python 3.6 issue as well as a python3.10 integration